modules/check_in/app/sidekiq/check_in/travel_claim_notification_job.rb

```diff
@@ -0,0 +1,213 @@
+    #
+    # @param opts [Hash] Options hash containing parameters for the notification
+    # @option opts [String] :mobile_phone The phone number to send the SMS to (required)
+    # @option opts [String] :appointment_date The appointment date in YYYY-MM-DD format (required)
+    # @option opts [String] :template_id The VaNotify template ID to use (required)
+    # @option opts [String] :claim_number The claim number to include in the message (required)
+    # @option opts [String] :facility_type The facility type ('oh' or 'cie'). Optional, defaults to 'cie' when nil.
+    # @return [void]
+    def perform(opts = {})
+      return unless validate_required_fields(opts)
+
+      parsed_date = parse_appointment_date(opts[:appointment_date])
+      return unless parsed_date
+
+      log_sending_travel_claim_notification(opts)
+      attempt_number = current_attempt_number
+
+      begin
+        va_notify_send_sms(opts, parsed_date)
+        StatsD.increment(Constants::STATSD_NOTIFY_SUCCESS)
+      rescue => e
+        log_send_sms_failure(attempt_number)
+        StatsD.increment(Constants::STATSD_NOTIFY_ERROR)
+        raise e
+      end
+    end
+
+    # Callback executed when all retries are exhausted
+    # @param job [Hash] The Sidekiq job hash containing job metadata
+    # @param ex [Exception] The exception that caused the job to fail
+    sidekiq_retries_exhausted do |job, ex|
+      CheckIn::TravelClaimNotificationJob.handle_error(job, ex)
+    end
+
+    # Handles errors after all retries have been exhausted
+    # Logs the error to Sentry and updates metrics based on the template type and facility
+    #
+    # @param job [Hash] The Sidekiq job hash containing job metadata
+    # @param ex [Exception] The exception that caused the job to fail
+    # @return [void]
+    def self.handle_error(job, ex)
+      opts = job.dig('args', 0) || {}
+
+      SentryLogging.log_exception_to_sentry(
+        ex,
+        {
+          phone_number: phone_last_four(opts),
+          template_id: opts&.dig(:template_id),
+          claim_number: opts&.dig(:claim_number)
+        },
+        { error: :check_in_va_notify_job, team: 'check-in' }
+      )
+
+      if FAILED_CLAIM_TEMPLATE_IDS.include?(opts&.dig(:template_id))
+        tags = if opts&.dig(:facility_type) == 'cie'
+                 Constants::STATSD_CIE_SILENT_FAILURE_TAGS
+               else
+                 Constants::STATSD_OH_SILENT_FAILURE_TAGS
+               end
+        StatsD.increment(Constants::STATSD_NOTIFY_SILENT_FAILURE, tags:)
+      end
+
+      StatsD.increment(Constants::STATSD_NOTIFY_ERROR)
+    end
+
+    # Extracts the last four digits of a phone number from a hash
+    # Removes any non-numeric characters before extracting the last four digits
+    #
+    # @param hash [Hash, nil] Hash containing a :mobile_phone key
+    # @return [String, nil] Last four digits of the phone number, or nil if not present
+    def self.phone_last_four(hash)
```

Comment:
I feel this function should be private

Reply:
@carlosfelix2 I agree that would make sense but the way the `sidekiq_retries_exhausted` logic runs it needs any methods called to be accessible from a class level because it does not re-instantiate. This is the only way to have it accessible when the job initially runs and when the death behavior is triggered. It seems like a bit of a shortcoming of `ActiveJob` but it seems preferable to having the method be defined twice for use in separate contexts.